app/src/main/java/com/example/chessclockk/screens/main/MainScreenContent.kt

```diff
@@ -0,0 +1,171 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Alarm
+import androidx.compose.material.icons.filled.Refresh
+import androidx.compose.material.icons.filled.Settings
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.tooling.preview.Preview
+import com.example.chessclockk.views.clock.ClockState
+import com.example.chessclockk.views.clock.ClockWidget
+import com.example.chessclockk.views.customtimebottomsheet.CustomTimeSetBottomSheetContent
+import com.example.chessclockk.views.playpause.PlayPause
+import com.example.chessclockk.views.playpause.PlayPauseState
+import com.example.chessclockk.views.restart.Restart
+import com.example.chessclockk.views.restart.RestartState
+import kotlinx.coroutines.launch
+
+@Composable
+@OptIn(ExperimentalMaterial3Api::class)
+fun MainScreenContent(
+    modifier: Modifier,
+    clockWhiteState: ClockState,
+    clockBlackState: ClockState,
+    playPauseState: PlayPauseState,
+    restartState: RestartState,
+    onCustomTimeSet: (String, String) -> Unit,
+    onCustomTimeSetClick: () -> Unit,
+    onSettingsClicked: () -> Unit
+) {
+
+    val sheetState = rememberModalBottomSheetState()
+    val coroutineScope = rememberCoroutineScope()
+    var showBottomSheet by remember { mutableStateOf(false) }
+    val settingsVisible by remember { mutableStateOf(false) }
+
+    if (showBottomSheet) {
+        ModalBottomSheet(
+            onDismissRequest = { showBottomSheet = false },
+            sheetState = sheetState
+        ) {
+            CustomTimeSetBottomSheetContent(
+                modifier = Modifier,
+                onSheetClose = { time, bonus ->
+                    coroutineScope.launch { sheetState.hide() }.invokeOnCompletion {
+                        if (!sheetState.isVisible) {
+                            showBottomSheet = false
+                        }
+                    }
+                    onCustomTimeSet(time, bonus)
+                }
+            )
+        }
+    }
+
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+            .fillMaxHeight()
+    ) {
+        ClockWidget(
+            modifier = Modifier.weight(1f),
```

Comment:
Podobno metody compose i ich argumenty potrafią się często zmienić. Dlatego w xtb wszędzie dajemy tutaj nazwane argumenty. To strasznie upierdliwe, ale zalecił to nasz spec od compose'a